### PR TITLE
feat(octavia.pools.members):  add IP instance - step 1

### PR DESCRIPTION
**Pull request by karim-ovh**
ref: MANAGER-10688

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/octavia-batch`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-10688
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Add ip instances first step on members tab

### CODE REVIEW

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/style.less [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.search-ip-instance {
+  margin-bottom: 17px;
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/style.less:2]: I think we have css classes to handle margin

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/constants.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const TRACKING_HIT_PREFIX = 'add';
```
[the remark this answers is not in the document]
Comment (karim-ovh) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/constants.js:1]: Exact i removed the constant file and trackCreateAction in routing file.

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    this.Alerter = Alerter;
+    this.OctaviaLoadBalancerMembersService = OctaviaLoadBalancerMembersService;
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    this.model = {};
+    this.instances = [];
+    this.displayedInstances = [];
+    this.model.checked = [];
```
[the remark this answers is not in the document]
Comment (karim-ovh) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js:14]: OK. I put this : 

    this.model = {
      checked: [],
    };

**Review by JacquesLarique — CHANGES_REQUESTED** [review]

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  computeIpV4Instance(instance) {
+    instance.ipAddresses
+      .filter((ipAddress) => ipAddress.version === 4)
+      .forEach((ipAddress) => {
+        this.instances.push({
+          label: `${instance.name} (${ipAddress.ip}) `,
+        });
+      });
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js:51]: Use only one loop
```suggestion
      .forEach((ipAddress) => {
        if (ipAddress.version === 4) {
          this.instances.push({
            label: `${instance.name} (${ipAddress.ip}) `,
          });
        }
      });
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  resetCurrentStep() {
+    this.currentStep = 0;
+  }
+
+  onSearchInstance(value) {
+    this.displayedInstances = this.instances.filter((instance) =>
+      instance.label.includes(value.toLowerCase()),
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js:60]: I think you should also use toLowerCase on instance.label
```suggestion
      instance.label.toLowerCase().includes(value.toLowerCase()),
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+                    >
+                        <oui-checkbox
+                            data-name="{{'instance-' + $index}}"
+                            data-model="$ctrl.model.checked[$index]"
+                        >
+                            <span data-ng-bind="instance.label"></span>
+                        </oui-checkbox>
+                    </span>
+                    <span data-ng-if="$ctrl.displayedInstances.length === 0">
+                        Aucune instance trouvée
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html:53]: No hard coded text

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                </oui-message>
+            </div>
+        </div>
+        <div class="row mb-2">
+            <div class="col-md-4">
+                <oui-search
+                    model="$ctrl.searchInstance"
+                    on-change="$ctrl.onSearchInstance(modelValue)"
+                    on-reset="$ctrl.onSearchReset()"
+                    placeholder="Rechercher une instance"
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html:34]: No hard coded text

**Review by JacquesLarique — CHANGES_REQUESTED** [review]

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  countIps(ipAddresses, type) {
+    return ipAddresses.reduce((accumulator, element) => {
+      if (element.type === type) {
+        return accumulator + 1;
+      }
+      return accumulator;
+    }, 0);
+  }
+
+  computeNewInstances(instance) {
```
[the remark this answers is not in the document]
Comment (karim-ovh) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js]: OK validated with Guillaume. We display all ips (public and private) but only version 4.

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js (hunk 5 above)
Comment (karim-ovh) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js:60]: Yes done.

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html (hunk 6 above)
Comment (karim-ovh) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html:53]: OK

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html (hunk 7 above)
Comment (karim-ovh) [reply packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html:34]: OK

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+  loadIpInstances() {
+    this.isLoading = true;
+    this.OctaviaLoadBalancerMembersService.getInstances(this.projectId)
+      .then((instances) => {
+        // For each instance, we create a new instance per ip address
+        // We create a new instance only for IPV4 ip address
+        angular.forEach(instances, (instance, index) => {
+          this.computeIpV4Instance(instance);
+          this.model.checked[index] = false;
```
Comment (qpavy) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/controller.js:28]: This should probably be on computeIpV4Instance function forEach to be in sync with instances array. To be fix on the next PR :)

**Review by qpavy — APPROVED** [review]

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        data-loading="$ctrl.isLoading"
+        data-description="{{ 'octavia_load_balancer_pools_detail_members_add_ip_instance_description' | translate }}"
+        data-prevent-next="true"
+        data-on-focus="$ctrl.resetCurrentStep()"
+    >
+        <div class="row mb-2">
+            <div class="col-md-8">
+                <oui-message
+                    type="warning"
+                    aria-close-button-label="Close"
```
Comment (JacquesLarique) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/detail/members/add-ip-instance/template.html:19]: No hard coded text

**Review by JacquesLarique — CHANGES_REQUESTED** [review]

**Pull request merged**